### ISSUE TITLE
Question about the KL divergence loss

### ISSUE THREAD

**Issue by marctimjen** [CompVis/stable-diffusion#849]
Hello

I hope someone can help me understand why the KL is calculated as:
0.5 * torch.sum(torch.pow(self.mean, 2) + self.var - 1.0 - self.logvar, dim=[1, 2, 3])

In the DiagonalGaussianDistribution listed here:
https://github.com/CompVis/stable-diffusion/blob/21f890f9da3cfbeaba8e2ac3c425ee9e998d5229/ldm/modules/distributions/distributions.py#L44

I am asking, because most loss functions for the VAE I can find use (-1 times this calculations) like this: 0.5 * torch.sum(-torch.pow(self.mean, 2) - self.var + 1.0 + self.logvar, dim=[1, 2, 3])

And I cannot see that we multiply by -1 in the contperceptual loss for instance: 

https://github.com/CompVis/stable-diffusion/blob/21f890f9da3cfbeaba8e2ac3c425ee9e998d5229/ldm/modules/losses/contperceptual.py#L83C57-L83C65

Thank you very much in advance :)

**Comment by marctimjen** [comment]
I found this material that does also have the loss on the form that is used here:

https://pyimagesearch.com/2023/10/02/a-deep-dive-into-variational-autoencoders-with-pytorch/

My confusion just happen because most papers write:

![image](https://github.com/CompVis/stable-diffusion/assets/79468558/56596092-42fd-4d2c-86b6-108ce5933337)

(From Bishops Deep learning)


And the original implementation of VAE:

![image](https://github.com/CompVis/stable-diffusion/assets/79468558/fcedb3a7-c70d-4a0d-a043-4e5654906419)

**Comment by tvelovraf** [comment]
According to original VAE paper this value (_resulting estimator_) is a value we want to **maximize**. Its first term is a KL divirgence multiplied by -1: `-D_kl`. KL devirgence is always positive, so `-D_kl` is negative.
When one talks about loss function, usually it means minimisation task. So if we want to maximize `-D_kl`, it's equivalent to minimization of `D_kl`. Therefore in loss we have this expression multiplied by -1
